apps/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -1016,6 +1016,18 @@ export class RushConfiguration {
         }
       }
     }
+
+    // Ensure the rush.json filename has the casing that exists on the filesystem
+    const correctlyCasedRushJsonFilename: string | undefined = Utilities.getCorrectlyCasedPath(rushJsonFilename);
```

Comment:
The design is that this constructor receives already validated inputs.

So maybe it would better to perform this normalization here, before we load the file:
https://github.com/microsoft/web-build-tools/blob/d377baa60cbefe16898ff0b6bdf80570a1efbf43/apps/rush-lib/src/api/RushConfiguration.ts#L309-L313

That might also enable you to throw a file-not-found exception rather than printing warning that will never get investigated.